### DISCUSSION TITLE
"Poetry build", unable to include "build/" directory

### DISCUSSION THREAD

**Discussion by cpolzer** [python-poetry/poetry#9840] [General] [answered]
Hello,

currently experimenting packaging sphinx projects including the resulting "objects.inv" files in packages and the rest of a documentation next to my actual python libs.

So to have the results of the sphinx "build/" included in my package, next to the sources.

Now to my question:
The sphinx build directory is called "build" in my poetry taskipy tasks, but i am not able to include the "build/" directory in the resulting package, its always "ignored".

```toml
# toml example
[tool.poetry]
name = "my-documentation"
version = "22.2.2"
description = "Some doumentation"
authors = ["Your Name <you@example.com>"]
readme = "README.md"
packages = [
    { include = "*" },
]
```

```text
# .gitignore example
build/
```

I tried following the documentation on [includes and excludes](https://python-poetry.org/docs/pyproject/#include-and-exclude), but then my wheel looks like the following, resulting that on the package consuming side, the build directory is just added to my venv and not inside my package contents in a flat structure.

```bash
poetry_sphinx_example-0.1.0-py3-none-any git:(main) ✗ tree 
.
├── build
│   ├── _sources
│   │   └── index.rst.txt
│   ├── _static
│   │   ├── alabaster.css
│   │   ├── basic.css
│   │   ├── custom.css
│   │   ├── doctools.js
│   │   ├── documentation_options.js
│   │   ├── file.png
│   │   ├── github-banner.svg
│   │   ├── language_data.js
│   │   ├── minus.png
│   │   ├── plus.png
│   │   ├── pygments.css
│   │   ├── searchtools.js
│   │   └── sphinx_highlight.js
│   ├── genindex.html
│   ├── index.html
│   ├── objects.inv
│   ├── search.html
│   └── searchindex.js
├── poetry-sphinx-example
│   └── source
│       ├── conf.py
│       └── index.rst
└── poetry_sphinx_example-0.1.0.dist-info
    ├── LICENSE
    ├── METADATA
    ├── RECORD
    └── WHEEL
````

Is there any hint as to how to solve this, or even better examples? Or shall i just use another directory name?

Regards!

EDIT:

Quickly setup a example repository with the current workaround i have: https://github.com/cpolzer/poetry-sphinx-example

I have to:
* rename my build dir to "_build" in order to not be ignored in full
* add a "__init__.py" to the _build/ directory just so that poetry will not fail with the message "X is not a package"

**Comment by radoering** [comment]
In case you only want to include in your sdist, something like

```
include = [
    { path = "build", format = "sdist" },
]
```

might work. If I remember correctly, `include` comes after `exclude`, where `.gitignore` files are automatically excluded. However, you have to be careful when using `include` with wheels. See #9691, which is only resolved in main and still an issue in 1.8.

**Reply by radoering** [reply, marked as answer]
On a second glance, I see your example:

```
packages = [
    { include = "source/", format = ["sdist", "wheel"], to = "poetry-sphinx-example" },
    # This is a hack to include the _build directory in the package: see the `build` task below
    { include = "_build/", format = ["sdist", "wheel"], to = "poetry-sphinx-example" },
]
```

Unfortunately, `to` is only supported in `packages` and not in `include`. I think you have already found the best workaround.
